### PR TITLE
Implement quota stats

### LINKED ISSUE #51: Support for quotas stats
It will be great to also collect quotas statistics from

`lctl get_param qmt.*.*.glb-*`:
```
qmt.testfs-QMT0000.dt-0x0.glb-grp=
global_pool0_dt_grp
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
qmt.testfs-QMT0000.dt-0x0.glb-prj=
global_pool0_dt_prj
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
- id:      1337
  limits:  { hard:               309200, soft:               307200, granted:              1025032, time:           1687277628 }
qmt.testfs-QMT0000.dt-0x0.glb-usr=
global_pool0_dt_usr
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
qmt.testfs-QMT0000.dt-ddn_ssd.glb-grp=
global_index_copy:
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
qmt.testfs-QMT0000.dt-ddn_ssd.glb-prj=
global_index_copy:
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
- id:      1337
  limits:  { hard:                    0, soft:                    0, granted:              1024004, time:      281474976710656 }
qmt.testfs-QMT0000.dt-ddn_ssd.glb-usr=
global_index_copy:
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
qmt.testfs-QMT0000.dt-pool_1.glb-grp=
global_index_copy:
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
qmt.testfs-QMT0000.dt-pool_1.glb-prj=
global_index_copy:
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
- id:      1337
  limits:  { hard:                    0, soft:                    0, granted:              1024004, time:      281474976710656 }
qmt.testfs-QMT0000.dt-pool_1.glb-usr=
global_index_copy:
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
qmt.testfs-QMT0000.md-0x0.glb-grp=
global_pool0_md_grp
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
qmt.testfs-QMT0000.md-0x0.glb-prj=
global_pool0_md_prj
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
- id:      1337
  limits:  { hard:                11000, soft:                10000, granted:                 1024, time:                    0 }
qmt.testfs-QMT0000.md-0x0.glb-usr=
global_pool0_md_usr
- id:      0
  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
```

`lctl get_param osd-ldiskfs.*.quota_slave.acct_*`:
```
osd-ldiskfs.testfs-MDT0000.quota_slave.acct_group=
grp_accounting:
- id:      0
  usage:   { inodes:                  396, kbytes:               529400 }
osd-ldiskfs.testfs-MDT0000.quota_slave.acct_project=
prj_accounting:
- id:      0
  usage:   { inodes:                  393, kbytes:               529396 }
- id:      1337
  usage:   { inodes:                    3, kbytes:                    4 }
osd-ldiskfs.testfs-MDT0000.quota_slave.acct_user=
usr_accounting:
- id:      0
  usage:   { inodes:                  396, kbytes:               529400 }
osd-ldiskfs.testfs-MDT0001.quota_slave.acct_group=
grp_accounting:
- id:      0
  usage:   { inodes:                  265, kbytes:               356672 }
osd-ldiskfs.testfs-MDT0001.quota_slave.acct_project=
prj_accounting:
- id:      0
  usage:   { inodes:                  265, kbytes:               356672 }
osd-ldiskfs.testfs-MDT0001.quota_slave.acct_user=
usr_accounting:
- id:      0
  usage:   { inodes:                  265, kbytes:               356672 }
```

### PR DESCRIPTION
**Pull request by RDruon**
Will fix #51

### CODE REVIEW

File: src/fixtures/valid/valid.txt [hunk 1]
```diff
@@ -276,3 +276,60 @@ req_active                1 samples [reqs] 1 1 1 1
...
+global_pool0_md_prj
+- id:      0
+  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
+- id:      1337
+  limits:  { hard:                11000, soft:                10000, granted:                 1024, time:                    0 }
+qmt.testfs-QMT0000.md-0x0.glb-usr=
+global_pool0_md_usr
+- id:      0
+  limits:  { hard:                    0, soft:                    0, granted:                    0, time:               604800 }
+ldlm.namespaces.filter-ai400-OST0001_UUID.resource_count=0
```
Comment (RDruon) [inline src/fixtures/valid/valid.txt:335]: If I remove, aka if `qmt.*` are the last values on the output, the parser fails with:
```
thread 'parser::tests::test_node_output' panicked at 'called `Result::unwrap()` on an `Err` value: Errors { position: 15297, errors: [Unexpected(Static("end of input")), Unexpected(Token('-')), Message(Static("while parsing qmt"))] }', src/parser.rs:153:14
```
It seems that's also the case with `jobstats`. Any idea how to solve it?

File: src/types.rs [hunk 2]
```diff
@@ -547,3 +559,54 @@ pub enum Record {
...
+    Usr,
+    Grp,
+    Prj,
+}
+
+#[derive(PartialEq, Eq, Debug, serde::Serialize, serde::Deserialize)]
+pub enum QuotaManager {
+    MDT,
+    OST,
+}
```
Comment (RDruon) [inline src/types.rs:612]:
```suggestion
```

Is it still needed?

File: src/quota/quota_parser.rs [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+pub(crate) fn quota_stats<I>() -> impl Parser<I, Output = Vec<QuotaStat>>
+where
+    I: Stream<Token = char>,
+    I::Error: ParseError<I::Token, I::Range, I::Position>,
+{
+    (
+        optional(newline()), // If quota stats are present, the whole yaml blob will be on a newline
+        many::<Vec<_>, _, _>(alpha_num().or(one_of("_-:".chars()))), // But yaml header might not be properly formatted, ignore it
```
Comment (jgrund) [inline src/quota/quota_parser.rs:70]: Example of this?

File: src/quota/quota_parser.rs [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+pub(crate) fn quota_stats_osd<I>() -> impl Parser<I, Output = Vec<QuotaStatOsd>>
+where
+    I: Stream<Token = char>,
+    I::Error: ParseError<I::Token, I::Range, I::Position>,
+{
+    (
+        optional(newline()), // If quota stats are present, the whole yaml blob will be on a newline
+        many::<Vec<_>, _, _>(alpha_num().or(one_of("_-:".chars()))), // But yaml header might not be properly formatted, ignore it
```
Comment (jgrund) [inline src/quota/quota_parser.rs:88]:
```suggestion
        many::<Vec<_>, _, _>(alpha_num())
```

File: src/quota/quota_parser.rs (hunk 3 above)
Comment (jgrund) [inline src/quota/quota_parser.rs:70]:
```suggestion
        many::<Vec<_>, _, _>(alpha_num())
```

File: src/quota/quota_parser.rs [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+pub(crate) fn quota_stats_osd<I>() -> impl Parser<I, Output = Vec<QuotaStatOsd>>
+where
+    I: Stream<Token = char>,
+    I::Error: ParseError<I::Token, I::Range, I::Position>,
+{
+    (
+        optional(newline()), // If quota stats are present, the whole yaml blob will start on a newline
+        many::<Vec<_>, _, _>(alpha_num().or(one_of("_-:".chars()))), // But yaml header might not be indented, ignore it
```
Comment (jgrund) [inline src/quota/quota_parser.rs:88]: We take a line of YAML here, but accept a subset of valid YAML...

This looks ok for now, but is a potential source of breakage going forwards.

File: src/quota/quota_parser.rs [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+pub(crate) fn quota_stats<I>() -> impl Parser<I, Output = Vec<QuotaStat>>
+where
+    I: Stream<Token = char>,
+    I::Error: ParseError<I::Token, I::Range, I::Position>,
+{
+    (
+        optional(newline()), // If quota stats are present, the whole yaml blob will start on a newline
+        many::<Vec<_>, _, _>(alpha_num().or(one_of("_-:".chars()))), // But yaml header might not be indented, ignore it
```
Comment (jgrund) [inline src/quota/quota_parser.rs:70]: We take a line of YAML here, but accept a subset of valid YAML...

This looks ok for now, but is a potential source of breakage going forwards.

**Review by jgrund — APPROVED** [review]

**Pull request merged**